From f6ee6ff7f92434ba85c258435ffc3431c14be6d2 Mon Sep 17 00:00:00 2001
From: Peleg Vigodny <peleg.vigodny@tandemg.com>
Date: Wed, 15 Jul 2020 12:55:23 +0300
Subject: [PATCH] [SW-12520] hls: install: host: deploy proj_hl

Deploy proj_hl directory as part of the release.

This directory has been developed and used by Accton to generate HLS FRU
binaries using frugen host tool.

It includes templates files specifying HLS FRU properties and scripts
used to generate binaries by converting these templates into JSON files
which are in turn provided to frugen utility.

Change-Id: I9910ff698288a6bedc6d870817f3aee48ca8ed8e
Signed-off-by: Peleg Vigodny <peleg.vigodny@tandemg.com>
---

diff --git a/package/openbmc/openbmc.mk b/package/openbmc/openbmc.mk
index b788f70..29a0ba4 100644
--- a/package/openbmc/openbmc.mk
+++ b/package/openbmc/openbmc.mk
@@ -6,6 +6,7 @@
 OBMC_PHOSPHOR_IMAGE = $(OPENBMC_IMAGE_DIR)/$(OBMC_PHOSPHOR_IMAGE_NAME)
 OBMC_PHOSPHOR_FULLCHIP_IMAGE = $(OPENBMC_IMAGE_DIR)/$(OBMC_PHOSPHOR_FULLCHIP_IMAGE_NAME)
 OPENBMC_FRUGEN_PATH = $(OPENBMC_IMAGE_DIR)/$(HLS_EEPROM_FRUGEN)
+OPENBMC_FRUGEN_HL_PROJ_PATH = $(OPENBMC_IMAGE_DIR)/$(HLS_FRUGEN_HL_PROJ)
 VERSION_FILE := $(OPENBMC_DEPLOY_DIR)/localversion-habanalabs
 
 define SET_VERSION_INFO
@@ -48,6 +49,9 @@
 	$(INSTALL) -m 0644 -D $(@D)/$(OBMC_PHOSPHOR_IMAGE) $(BINARIES_DIR)/$(OBMC_PHOSPHOR_IMAGE_NAME)
 	$(INSTALL) -m 0644 -D $(@D)/$(OBMC_PHOSPHOR_FULLCHIP_IMAGE) $(BINARIES_DIR)/$(OBMC_PHOSPHOR_FULLCHIP_IMAGE_NAME)
 	$(INSTALL) -m 0755 -D $(@D)/$(OPENBMC_FRUGEN_PATH) $(BINARIES_DIR)
+	$(INSTALL) -d $(BINARIES_DIR)/$(HLS_FRUGEN_HL_PROJ)
+	find $(@D)/$(OPENBMC_FRUGEN_HL_PROJ_PATH) -name '*.sh' -exec $(INSTALL) -m 0755 -D {} $(BINARIES_DIR)/$(HLS_FRUGEN_HL_PROJ) \;
+	find $(@D)/$(OPENBMC_FRUGEN_HL_PROJ_PATH) ! -name '*.sh' -exec $(INSTALL) -m 0644 -D {} $(BINARIES_DIR)/$(HLS_FRUGEN_HL_PROJ) \;
 	$(INSTALL) -m 0755 -D ${BR2_EXTERNAL_HABANALABS_PATH}/package/openbmc/install-hls.sh $(BINARIES_DIR)/install-hls.sh
 	$(INSTALL) -m 0755 -D ${BR2_EXTERNAL_HABANALABS_PATH}/package/openbmc/install-hls-curr.sh $(BINARIES_DIR)/install-hls-curr.sh
 	$(INSTALL) -m 0755 -D ${BR2_EXTERNAL_HABANALABS_PATH}/package/openbmc/install-hls-legacy.sh $(BINARIES_DIR)/install-hls-legacy.sh
diff --git a/scripts/make/hls/Makefile b/scripts/make/hls/Makefile
index de92ed9..16de29a 100644
--- a/scripts/make/hls/Makefile
+++ b/scripts/make/hls/Makefile
@@ -2,6 +2,7 @@
 export OPENBMC_ENV_FLAGS = meta-habana/meta-hls/conf
 export OPENBMC_IMAGE_NAME = flash-hls
 export HLS_EEPROM_FRUGEN = frugen
+export HLS_FRUGEN_HL_PROJ = proj_hl
 export OBMC_PHOSPHOR_IMAGE_BASENAME = obmc-phosphor-image-hls
 export OBMC_PHOSPHOR_IMAGE_NAME = $(OBMC_PHOSPHOR_IMAGE_BASENAME).static.mtd.tar
 export OBMC_PHOSPHOR_FULLCHIP_IMAGE_NAME = $(OBMC_PHOSPHOR_IMAGE_BASENAME).static.mtd.all.tar
@@ -49,6 +50,7 @@
 	@sudo mkdir -p "$(HLS_RELEASE_DIR)"
 	@sudo cp -p $(IMAGES_DIR)/$(HLS_PKG_IMG_NAME) $(HLS_RELEASE_DIR)
 	@sudo cp -p $(IMAGES_DIR)/$(HLS_EEPROM_FRUGEN) $(HLS_RELEASE_DIR)
+	@sudo cp -rp $(IMAGES_DIR)/$(HLS_FRUGEN_HL_PROJ) $(HLS_RELEASE_DIR)
 	@echo "$(EMBD_FULL_VER)" | sudo tee $(HLS_RELEASE_DIR)/hl-fw-version.txt
 	@sudo ln -nfs $$(basename $(HLS_RELEASE_DIR)) $(HLS_RELEASE_DIR_BASE)/$(1)-release
 	@echo -n "The HLS image version $(EMBD_FULL_VER), installed to "
